pylutron_caseta/pairing.py

```diff
@@ -0,0 +1,192 @@
+"""Guide the user through pairing and save the necessary files."""
+
+import json
+import logging
+import socket
+import ssl
+import tempfile
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.x509.oid import NameOID  # type: ignore
+
+from .assets import LAP_CA_PEM, LAP_CERT_PEM, LAP_KEY_PEM
+
+LOGGER = logging.getLogger(__name__)
+
+SOCKET_TIMEOUT = 10
+BUTTON_PRESS_TIMEOUT = 180
+CERT_SUBJECT = x509.Name(
+    [x509.NameAttribute(NameOID.COMMON_NAME, "get_lutron_cert.py")]
```

Comment:
We probably want a different subject name here, or even a way for the caller to provide the subject name.

Reply:
Adjusted to `pylutron_caseta`